modules/ppcp-onboarding/src/Render/OnboardingRenderer.php

```diff
@@ -83,9 +94,17 @@ public function get_signup_link( bool $is_production, array $products ) {
 			->with_products( $products )
 			->data();
 
+		$environment = $is_production ? 'production' : 'sandbox';
+		$product     = isset( $data['products']['PPCP'] ) ? 'ppcp' : 'express-checkout';
+		if ( $this->cache->has( $environment . '-' . $product ) ) {
+			return $this->cache->get( $environment . '-' . $product );
+		}
+
 		$url = $is_production ? $this->production_partner_referrals->signup_link( $data ) : $this->sandbox_partner_referrals->signup_link( $data );
 		$url = add_query_arg( $args, $url );
 
+		$this->cache->set( $environment . '-' . $product, $url, 3 * MONTH_IN_SECONDS );
```

Comment:
3 months seems a lot, is that how long the link supposed to work?

Reply:
yup, I asked PayPal and they said that referral link should be valid for about 3 months.